pkg/osutil/interrupt_unix.go

```diff
@@ -69,6 +69,11 @@ func HandleInterrupts() {
 			os.Exit(0)
 		}
 		syscall.Kill(pid, sig.(syscall.Signal))
+
+		// unlock exit lock since the go runtime doesn't reinstall
```

Comment:
It's like the comment says: the golang runtime won't always reset the signal handler to default before sending the signal back to itself, so it won't always terminate and instead it will keep running.

Run etcd via:
```sh
#!/bin/bash
./bin/etcd &
wait $!
```

Try to SIGINT it:
```sh
$ kill -SIGINT `pgrep etcd`
```

and etcd doesn't die